DominantColor/Shared/KMeans.swift

```diff
@@ -71,7 +71,12 @@ func kmeans<T : ClusteredType>(
         previousError = error
     } while abs(error - previousError) > threshold
     
-    return map(Zip2(centroids, clusterSizes)) { Cluster(centroid: $0, size: $1) }
+        var clusters = [Cluster<T>]()
+        for (first, second) in Zip2Sequence(centroids, clusterSizes) {
```

Comment:
Is there a reason why `Zip2Sequence.map` can't be used here instead of the for loop?